ehrapy/preprocessing/_imputation.py

```diff
@@ -382,10 +341,10 @@ def miss_forest_impute(
         _warn_imputation_threshold(adata, list(adata.var_names), threshold=warning_threshold)
     elif isinstance(var_names, dict):
         _warn_imputation_threshold(adata, var_names.keys(), threshold=warning_threshold)  # type: ignore
-    elif isinstance(var_names, list):
+    elif isinstance(var_names, Iterable) and all(isinstance(item, str) for item in var_names):
```

Comment:
Why do we need the second check? You surely have a good reason but I'd just like to understand. It's another operation so I wonder whether we need it.

Reply:
Because while we can check that `var_names` is an `Iterable`, we cannot specifically check if it is a `Iterable[str]` (AFAIK)